bootstrap/config/config.go

```diff
@@ -458,7 +458,7 @@ func (cp *Processor) listenForChanges(serviceConfig interfaces.Configuration, co
 		updateStream := make(chan interface{})
 		defer close(updateStream)
 
-		configClient.WatchForChanges(updateStream, errorStream, serviceConfig.EmptyWritablePtr(), writableKey)
+		go configClient.WatchForChanges(updateStream, errorStream, serviceConfig.EmptyWritablePtr(), writableKey)
```

Comment:
This is redundant.
The implementation of configClient.WatchForChanges is already async. 
https://github.com/edgexfoundry/go-mod-configuration/blob/main/internal/pkg/consul/client.go#L250

Reply:
@lenny-intel this is calling the abstraction interface fucnction, so we don't know whether the implementation is async.
also if something wrong happens in this area, the process will be stuck
https://github.com/edgexfoundry/go-mod-configuration/blob/a360551635e96eb79b5c7c3dd45f86fa62f7369c/internal/pkg/consul/client.go#L235-L245

From my opinion, I thought the redundant is in the consul/client.go 

In our use case, we would like to establish the message bus connection first, and there is no way to return the error if the connection cannot be established.